### PR TITLE
feature: Add apt package to universe

### PR DESCRIPTION
**Pull request by cpuguy83**
This packages is fairly similar in interface to the alpine package.

Some specific features of this over a standard ubuntu install:

1. Configured to cache packages
2. Stores apt cache in cache mounts
3. For the purposes of installing packages specified, it undoes some of
   the built-in apt configs that help keep the image smaller since these
   are not needed when using cache mounts and messes up the cache.
4. Some optimizations so that `apt-get update` is only run when the apt
   cache is older than 60 minutes (or non-existent).

### CODE REVIEW

**Comment by cpuguy83** [comment]
I cannot reproduce the linting error

File: pkg/universe.dagger.io/ubuntu/test/test.bats
Comment (TomChv) [inline pkg/universe.dagger.io/ubuntu/test/test.bats:2]: `../../../../bats_helpers` since your package will be in `universe.dagger.io/x/<your email>`

File: pkg/universe.dagger.io/ubuntu/ubuntu.cue
Comment (TomChv) [inline pkg/universe.dagger.io/ubuntu/ubuntu.cue:43]: Nit: Avoid if, prefer [disjunction](https://docs.dagger.io/1226/coding-style#favor-disjunctions-over-if-conditions)

File: pkg/universe.dagger.io/ubuntu/ubuntu.cue
Comment (TomChv) [inline pkg/universe.dagger.io/ubuntu/ubuntu.cue:64]: Nit: Avoid if, prefer [disjunction](https://docs.dagger.io/1226/coding-style#favor-disjunctions-over-if-conditions)

File: pkg/universe.dagger.io/ubuntu/ubuntu.cue
Comment (TomChv) [inline pkg/universe.dagger.io/ubuntu/ubuntu.cue:83]: Nit: `type: "fs"` is infered, you could remove it

**Review by TomChv — COMMENTED** [review]
Thanks for your contribution.

Experimental package should be in `universe.dagger.io/x/<your email>`, you can find example [here](https://github.com/dagger/dagger/tree/main/pkg/universe.dagger.io/x).

I left nit comments to improve it.

Fix the linter, put the package in `universe.dagger.io` and we can merge that one after another pair of eyes look at your PR :D

**Comment by cpuguy83** [comment]
Updated.

- Moved to `x/<email>`
- Changed package from `ubuntu` to `apt`
- Now includes an `#Install` (instead of `#Build`) action which is a build step that can be inserted into a `docker.#Build` to install packages.

File: pkg/universe.dagger.io/x/cpuguy83@gmail.com/apt/install.cue
Comment (sipsma) [inline pkg/universe.dagger.io/x/cpuguy83@gmail.com/apt/install.cue:105]: 1. This is neat
2. Do you think it would make sense to make the `60` configurable? In case someone wants to force a re-run of apt update? Not necessary if it's hard for whatever reason, just a thought in case it's easy.

**Review by sipsma — COMMENTED** [review]
Just quick nits and suggestions, looks awesome! I have a vague, unpleasant memory of trying to add all these types optimizations while writing a Dockerfile several years ago, great to see it will be so easy + reusable now.

File: pkg/universe.dagger.io/x/cpuguy83@gmail.com/apt/install.cue
Comment (cpuguy83) [reply pkg/universe.dagger.io/x/cpuguy83@gmail.com/apt/install.cue]: Added.

File: pkg/universe.dagger.io/x/cpuguy83@gmail.com/apt/install.cue
Comment (cpuguy83) [reply pkg/universe.dagger.io/x/cpuguy83@gmail.com/apt/install.cue]: Done.

**Comment by cpuguy83** [comment]
Seems like there's some issues with gha cache.

**Comment by sipsma** [comment]
> Seems like there's some issues with gha cache.

Yeah I'm taking a look at the Gitpod issue first and I'll look at the GHA one next. None of it has to do with your PR though

**Review by shykes — APPROVED** [review]
Pro-actively approving so we can merge as soon as tests pass. Thanks for the contribution Brian!

I think we should consider fast-tracking this to an official universe package too, given the API similarity to alpine... Seems relatively low risk.

**Comment by sipsma** [comment]
@cpuguy83 the Gitpod issue has been fixed in main, once you rebase this should be good to go. I confirmed in the jaeger traces that the timeouts in the test were just 429's from GHA, so they should pass on retries.

**Review by sipsma — APPROVED** [review]

**Pull request merged**
